### PR TITLE
ENH: Accelerate image list keys generation

### LINKED ISSUE #1987: Provide public interface for skipping inline page images
## Explanation

I want to extract all images from a page, but omit inline images as they are not really useful in my case and just generate overhead (2 ms without and 29 s with inline images for one page with a dotted table which has 24643 inline images, but no "real" images).

## Code Example

For now, I am basically exploiting https://github.com/py-pdf/pypdf/blob/c2a741e968da1b4d7195a7924ba62a9445f825e4/pypdf/_page.py#L478 which does not seem to be a clean solution:

```python
from pypdf import PdfReader

reader = PdfReader(path)
for page in reader.pages:
    page.inline_images = dict()  # Avoid loading inline images.
    for image in page.images:
        print(image)
```

### PR DESCRIPTION
**Pull request by pubpub-zz**
closes #1987

### CODE REVIEW

**Comment by pubpub-zz** [comment]
@MartinThoma 
I've got 2 mypy errors I do not understand Can you have a look please 😘

**Comment by pubpub-zz** [comment]
@MartinThoma 
I've found a nice fix. Now it's all your 😀

**Comment by MartinThoma** [comment]
I've tested it with https://github.com/py-pdf/pypdf/files/12160419/table_redacted.pdf and now I get:

```
Traceback (most recent call last):
  File "/home/moose/Github/py-pdf/pypdf/sample-files/foo.py", line 20, in <module>
    run("table_redacted.pdf")
  File "/home/moose/Github/py-pdf/pypdf/sample-files/foo.py", line 13, in run
    for image in page.images:
  File "/home/moose/Github/py-pdf/pypdf/pypdf/_page.py", line 2633, in __iter__
    yield self[i]
          ~~~~^^^
  File "/home/moose/Github/py-pdf/pypdf/pypdf/_page.py", line 2629, in __getitem__
    return self.get_function(lst[index])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/moose/Github/py-pdf/pypdf/pypdf/_page.py", line 532, in _get_image
    return self.inline_images[id]
           ~~~~~~~~~~~~~~~~~~^^^^
KeyError: '~0~'
```

**Comment by stefan6419846** [comment]
Which code did you use for testing? Did you remove `page.inline_images = dict()`?

**Comment by MartinThoma** [comment]
Thank you - I forgot that :see_no_evil:

**Comment by MartinThoma** [comment]
Before (current main):

* 4.24s: 009-pdflatex-geotopo/GeoTopo.pdf
* 2.88s: 009-pdflatex-geotopo/GeoTopo-komprimiert.pdf

With this PR:

* 2.01s: 009-pdflatex-geotopo/GeoTopo.pdf
* 0.44s: 009-pdflatex-geotopo/GeoTopo-komprimiert.pdf

Good work :tada:

**Comment by pubpub-zz** [comment]
> I've tested it with https://github.com/py-pdf/pypdf/files/12160419/table_redacted.pdf and now I get:
> 
> ```
> Traceback (most recent call last):
>   File "/home/moose/Github/py-pdf/pypdf/sample-files/foo.py", line 20, in <module>
>     run("table_redacted.pdf")
>   File "/home/moose/Github/py-pdf/pypdf/sample-files/foo.py", line 13, in run
>     for image in page.images:
>   File "/home/moose/Github/py-pdf/pypdf/pypdf/_page.py", line 2633, in __iter__
>     yield self[i]
>           ~~~~^^^
>   File "/home/moose/Github/py-pdf/pypdf/pypdf/_page.py", line 2629, in __getitem__
>     return self.get_function(lst[index])
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/moose/Github/py-pdf/pypdf/pypdf/_page.py", line 532, in _get_image
>     return self.inline_images[id]
>            ~~~~~~~~~~~~~~~~~~^^^^
> KeyError: '~0~'
> ```

can you clarify the code you are using ? ` page.inline_images = dict()` is normally not required

**Comment by MartinThoma** [comment]
Yes, I was adding `page.inline_images = dict()`. That was leading to the error. However, I would not consider this a blocker as this is modifying pypdf behavior in an unexpected way.

I want to have a final look after work, but so far it seems like a great improvement. I'll likely merge it as it is :-)

File: pypdf/_page.py [hunk 1]
```diff
@@ -894,6 +906,23 @@ def _add_transformation_matrix(
         )
         return contents
 
+    def _get_contents_as_bytes(self) -> Optional[bytes]:
+        """
+        Return the page contents as bytes .
+
+        Returns:
+            The ``/Contents`` object as bytes, or ``None`` if it doesn't exist.
```
Comment (MartinThoma) [inline pypdf/_page.py:914]:
```suggestion
        Return the page contents as bytes.

        Returns:
            The ``/Contents`` object as bytes, or ``None`` if it doesn't exist.
```

**Pull request merged**